lib/chargebee/nativeRequest.rb

```diff
@@ -73,7 +73,7 @@ def self.handle_response(response, headers)
       # decompress the response. Therefore, we need to manually handle decompression
       # based on the 'Content-Encoding' header in the response.
       # https://github.com/ruby/ruby/blob/19c1f0233eb5202403c52b196f1d573893eacab7/lib/net/http/generic_request.rb#L82
-      if headers.keys.any? { |k| k.downcase == 'accept-encoding' } && rheaders[:content_encoding] == 'gzip' && rbody && !rbody.empty?
+      if rheaders[:content_encoding] == 'gzip' && rbody && !rbody.empty?
```

Comment:
```suggestion
      if !headers.keys.any? { |k| k.downcase == 'accept-encoding' } && rheaders[:content_encoding] == 'gzip' && rbody && !rbody.empty?
```

What do you think about this, this should actually be an implementation of what is described by NET::HTTP as I showed above.

Reply:
Hi @eduAntequera, thanks for the detailed explanation. Let us ensure backward compatibility and check for any failure cases. If none, we will roll this out. Does that work?

Reply:
that works :100: